CHANGELOG.adoc

```diff
@@ -33,6 +33,42 @@ More existing APIs will be migrated to this new common pattern.
  - https://github.com/eclipse-sirius/sirius-web/issues/3634[#3634] [sirius-web] Remove `GraphQLNodeStyleFragment` from `NodeTypeRegistry`, you can specify additional fields to be retreived by the subscription using the `DocumentTransform` GraphQL API and the `apolloClientOptionsConfigurersExtensionPoint` extension point
  - https://github.com/eclipse-sirius/sirius-web/issues/3641[#3641] [core] `IRepresentationMetadataProvider` now defined a single method `Optional<RepresentationMetadata> getMetadata(String representationId)` instead of separate `canHandle` and `handle` methods.
 
+ - https://github.com/eclipse-sirius/sirius-web/issues/2759[#2759] [diagram] Update some API relative to the diagram:
+
+ * The `NodeTool#selectionDescription` attribute of type `SelectionDescription` has been renamed into `dialogDescription` and is now of type `DialogDescription`.
+ * The `SelectionDescription` type has been renamed into `SelectionDialogDescription` and inherit from `DialogDescription`.
+ * The `InvokeSingleClickOnDiagramElementToolInput#selectedObjectId` has been replaced by a `variables: [ToolVariable!]!` to make it possible to provide any kind of variable to the backend context.
+
+```
+ input InvokeSingleClickOnDiagramElementToolInput {
+  id: ID!
+  editingContextId: ID!
+  representationId: ID!
+  variables: [ToolVariable!]!
+  diagramElementId: ID!
+  startingPositionX: Float!
+  startingPositionY: Float!
+  toolId: ID!
+}
+```
+
+ * The `SingleClickOnDiagramElementTool#dialogDescriptionId` has been replaced by a `dialog: Dialog` field to provide the information about the dialog that the frontend needs to retrieve in the extension point.
```

Comment:
Why isn't that a XXXDescription? Could you please use something like `ToolDialogDescription` or `DiagramToolDialogDescription` in order not to use such a common name which may be used elsewhere in the future.

Why the need for a whole concept instead of of the identifier?

Reply:
It idea was to own both the id and the information about the dialog type. We need to change the id to own the information about the dialog type if we want to keep only a dialogDescriptionId field?

Reply:
I still don't know why we would need both this two pieces of information. Why isn't an identifier enough?

Reply:
I am modifying the way we generate this identifier to have something similar to the other representation: siriusComponents://dialogDescription?kind=selectionDialogDescription etc"

Reply:
done